src/atoms/TextInput/TextInput.native.js

```diff
@@ -0,0 +1,333 @@
+import View from '../View';
+import isEmpty from '../../_helpers/isEmpty';
+import Size from '../Size';
+
+const IS_ANDROID = Platform.OS === 'android';
+
+const styles = {
+  textInput: {
+    padding({ variant, hasLeftIcon, hasPrefix, hasText }) {
+      const paddingTop = IS_ANDROID ? '8px' : '0px';
+      const paddingRight = '0px';
+      const paddingBottom = IS_ANDROID ? '0px' : hasText ? '4px' : '2px';
+      const paddingLeft = variant === 'outlined' || hasLeftIcon || hasPrefix ? '0px' : '8px';
+      // iOS & Android need different paddings
+      return `${paddingTop} ${paddingRight} ${paddingBottom} ${paddingLeft}`;
+    },
+    fontSize({ theme }) {
+      return theme.fonts.size.medium;
+    },
+    lineHeight({ theme }) {
+      return getLineHeight(theme, 'medium');
+    },
+    fontFamily({ theme }) {
+      return theme.fonts.family.lato.regular;
+    },
+    color({ theme, disabled }) {
+      if (disabled) {
+        return theme.colors.shade[940];
+      } else {
+        return theme.colors.shade[980];
+      }
+    },
+  },
+  fillContainer: {
+    backgroundColor({ variant, isFocused, theme, disabled }) {
+      if (variant === 'outlined') {
+        return 'transparent';
+      } else if (disabled) {
+        return theme.colors.tone[930];
+      } else if (isFocused) {
+        return theme.colors.tone[940];
+      } else {
+        return theme.colors.tone[930];
+      }
+    },
+  },
+  inputContainer: {
+    width({ size }) {
+      switch (size) {
+        case 'block':
+          return '100%';
+        case 'small':
+          return '160px';
+        case 'medium':
+          return '240px';
+        default:
+          return '240px';
+      }
+    },
+  },
+};
+
+const InputContainer = styled(View)`
+  background-color: transparent;
+`;
+
+const StyledInput = styled(NativeTextInput)`
+  padding: ${styles.textInput.padding};
+  font-size: ${styles.textInput.fontSize};
+  line-height: ${styles.textInput.lineHeight};
+  font-family: ${styles.textInput.fontFamily};
+  color: ${styles.textInput.color};
+`;
+
+const FillContainer = styled(View)`
+  background-color: ${styles.fillContainer.backgroundColor};
+  border-top-left-radius: 2px;
+  border-top-right-radius: 2px;
+`;
+
+const getAccessoryConfig = ({ errorText, prefix, suffix, iconLeft, iconRight }) => {
+  const hasError = !isEmpty(errorText);
+  const hasPrefix = !isEmpty(prefix);
+  const hasSuffix = !isEmpty(suffix);
+  const hasLeftIcon = !hasPrefix && !isEmpty(iconLeft);
+  const hasRightIcon = !hasSuffix && !isEmpty(iconRight);
+  return { hasError, hasPrefix, hasSuffix, hasLeftIcon, hasRightIcon };
+};
+
+const TextInput = ({
+  placeholder,
+  onChange,
+  helpText,
+  errorText,
+  variant,
+  prefix,
+  suffix,
+  disabled,
+  children,
+  iconLeft,
+  iconRight,
+  maxLength,
+  showCharacterCount,
+  label,
+  testID,
+  labelPosition,
+  size,
+}) => {
+  const theme = useContext(ThemeContext);
+  const [isFocused, setFocused] = useState(false);
```

Comment:
it should be `isFocused, setIsFocused` or rename it entirely to `focused, setFocused`